### PR TITLE
BUG: #509 remove excessive indexing & adjust portfolio value join & TZ awareness

### PR DESCRIPTION
**Pull request by h55nick**
NOTE: I did not add or adjust tests and have only tested this with my local setup. I do not endorse the merging of this without further testing. Even opening this PR is against my better judgment as a developer.

This fixed: https://github.com/quantopian/pyfolio/issues/509 for my local usage. 

cc/ @twiecki

### CODE REVIEW

File: pyfolio/round_trips.py [hunk 1]
```diff
@@ -260,15 +252,21 @@ def extract_round_trips(transactions,
...
-        roundtrips['date'] = roundtrips.close_dt.apply(lambda x:
-                                                       x.replace(hour=0,
-                                                                 minute=0,
-                                                                 second=0))
-
-        tmp = roundtrips.join(pv, on='date', lsuffix='_')
-
-        roundtrips['returns'] = tmp.pnl / tmp.portfolio_value
-        roundtrips = roundtrips.drop('date', axis='columns')
+        roundtrips['date'] = roundtrips.close_dt.apply(
```
Comment (twiecki) [inline pyfolio/round_trips.py]: Does `roundtrips.close_dt.normalize()` work?

Comment (h55nick) [reply pyfolio/round_trips.py]: I'd imagine not, as close_dt would return a Series.
You could `.normalize()` within the lambda? But I still think you'd run into a tz-aware issue as my reading of normalize docs, it does not change the awareness of the TS.

Assuming the problem here is the dep pytz (and generally that this lambda is pretty lacking style in general)

Comment (twiecki) [reply pyfolio/round_trips.py]: OK, there's probably a reason for this pattern here.

**Comment by twiecki** [comment]
What was the specific bug that's fixed here?

**Comment by twiecki** [comment]
Also note that we enforce pep8:
```
pyfolio/round_trips.py:128:80: E501 line too long (85 > 79 characters)
pyfolio/round_trips.py:129:80: E501 line too long (86 > 79 characters)
pyfolio/round_trips.py:132:68: E201 whitespace after '{'
pyfolio/round_trips.py:132:80: E501 line too long (119 > 79 characters)
pyfolio/round_trips.py:266:1: W293 blank line contains whitespace
pyfolio/round_trips.py:268:1: W293 blank line contains whitespace
pyfolio/round_trips.py:411:1: W391 blank line at end of file
```

File: pyfolio/round_trips.py (hunk 1 above)
Comment (h55nick) [reply pyfolio/round_trips.py]: which pattern exactly?

The only real add as the tz-aware a reordering and then a removal of: `t = t.reset_index()` which is what caused the collision.

Comment (twiecki) [reply pyfolio/round_trips.py]: Just saying, `.normalize()` and `.tz_localize('utc')` should to the same thing, just vectorized.

**Comment by twiecki** [comment]
Closing due to inactivity from submitter.

**Pull request closed without merging**